vespalib/src/vespa/vespalib/datastore/array_store.h

```diff
@@ -43,26 +44,54 @@ class ArrayStore : public ICompactable
     using RefType = RefT;
     using SmallBufferType = typename TypeMapperT::SmallBufferType;
     using TypeMapper = TypeMapperT;
+    struct no_vector { };
+
+    template <class, class = void>
+    struct check_dynamic_buffer_type_member {
+        static constexpr bool value = false;
+        using vector_type = no_vector;
+    };
+
+    template <class T>
+    struct check_dynamic_buffer_type_member<T, std::void_t<typename T::DynamicBufferType>> {
+        static constexpr bool value = true;
+        using vector_type = std::vector<typename T::DynamicBufferType>;
+    };
+
+    static constexpr bool has_dynamic_buffer_type = check_dynamic_buffer_type_member<TypeMapper>::value;
+    using DynamicBufferTypeVector = check_dynamic_buffer_type_member<TypeMapper>::vector_type;
 private:
     uint32_t                     _largeArrayTypeId;
     uint32_t                     _maxSmallArrayTypeId;
     size_t                       _maxSmallArraySize;
     DataStoreType                _store;
     TypeMapper                   _mapper;
     std::vector<SmallBufferType> _smallArrayTypes;
+    [[no_unique_address]] DynamicBufferTypeVector _dynamicArrayTypes;
```

Comment:
Please update the class description of ArrayStore to reflect the new functionality.